tensorflow/contrib/tensorrt/segment/segment.cc

```diff
@@ -236,9 +362,22 @@ tensorflow::Status SegmentGraph(
   // Collect the segments/subgraphs. Each subgraph is represented by a
   // set of the names of the nodes in that subgraph.
   std::unordered_map<string, std::set<string>> sg_map;
+  std::unordered_map<string, std::set<string>> device_maps;
   for (auto& u : node_segments) {
     if ((u.Value() != nullptr) && (u.ParentValue() != nullptr)) {
       sg_map[u.ParentValue()->name()].insert(u.Value()->name());
+      auto tf_node = u.Value()->tf_node();
+      if (tf_node->has_assigned_device_name()) {
```

Comment:
I realized that we need to accommodate the existing python graph conversion api, in that case please see my comment below, by running the original segmentation algorithm multiple times, one for each subgraph with different device assignment, we don't need the logic here and it can minimize the change to segment.h/cc.

Reply:
Actually, first condition is false and second is true since graph is converted to graphdef before it is passed to optimization passes which converts assigned devices to requested devices. I keep it there for potential future updates.